addons/project-folders/userscript.js

```diff
@@ -0,0 +1,191 @@
+export default async function ({ addon, global, console, msg }) {
+  let projectColumns;
+  projectColumns = await addon.tab.waitForElement("ul.media-list", {
+    markAsSeen: true,
+  });
+
+  let foldersJSON;
+  const projectTab = document.querySelectorAll("li.first")[0];
+  projectTab.childNodes[0].onclick = async () => {
+    projectColumns = await addon.tab.waitForElement("ul.media-list", {
+      markAsSeen: true,
+    });
+    createFolderAreaAndButton();
+    foldersJSON = await load();
+  };
+
+  if (document.querySelectorAll("div.folders-container")[0] === undefined) {
+    createFolderAreaAndButton();
+    foldersJSON = await load();
+  }
+
+  async function getProjectDetails(projectID) {
+    const token = await addon.auth.fetchXToken();
+    const projectDetails = await (
+      await fetch(`https://api.scratch.mit.edu/projects/${projectID}/`, {
+        headers: {
+          "content-type": "application/json",
+          "x-csrftoken": addon.auth.crsfToken,
+          "x-token": token,
+        },
+      })
+    ).json();
+
+    return projectDetails;
+  }
+
+  async function load() {
+    const foldersJSON = { folders: [] };
```

Comment:
Why this structure? Couldn't this just be an array instead of being nested in an object?

Reply:
I have changed it into an array in the latest commit.